src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -653,8 +654,13 @@ namespace Microsoft.FSharp.Core
 
             let inline Array2DZeroCreate (n:int) (m:int) = (# "newarr.multi 2 !0" type ('T) n m : 'T[,] #)
             let GetArray2DSub (src: 'T[,]) src1 src2 len1 len2 =
-                let len1 = (if len1 < 0 then 0 else len1)
-                let len2 = (if len2 < 0 then 0 else len2)
+                let mutable len1 = len1
```

Comment:
That's not prohibited as in we throw exception, but if you get a slice with 0 as any one of the dimensions, the user should get an empty array of size (0,0), instead of the old behavior where if they specified a slice dimension of (x, 0) they got a slice of size (x, 0), but if they specified a slice dimension of (0, x) they got a slice of (0, 0).

Reply:
@nelson, the new code returns an empty multi-dimension array if any of the dimensions are empty.  This is a change from the original code which would return an array containing an empty array only for the empty slice.

I think I said that right.